### PR TITLE
test: better test coverage

### PR DESCRIPTION
**Pull request by fattybenji**
Adds some coverage mainly on cli and vcs_helpers

### CODE REVIEW

File: tests/test_vcs_helpers.py [hunk 1]
```diff
@@ -14,6 +16,12 @@ def mock_git(mocker):
     return mocker.patch('semantic_release.vcs_helpers.repo.git')
 
 
+@mock.patch('semantic_release.vcs_helpers.repo', None)
+def test_check_repo():
```
Comment (relekang) [inline tests/test_vcs_helpers.py]: This test name should reflect what it checks.

File: tests/test_pypi.py [hunk 2]
```diff
@@ -39,3 +40,7 @@ def test_upload_with_custom_path(self, mock_run):
         self.assertEqual(args[2], mock.call(
             'twine upload -u username -p password  custom-dist/*'),)
         self.assertEqual(args[3], mock.call('rm -rf custom-dist'))
+
+    def test_raises_error(self):
```
Comment (relekang) [inline tests/test_pypi.py]: This test name should reflect what it checks, something like `test_raises_error_when_missing_credentials`.

File: tests/test_cli.py [hunk 3]
```diff
@@ -341,9 +473,81 @@ def test_changelog_should_call_functions(mocker, runner):
...
+
+
+def test_changelog_raises_exception(mocker):
+    mocker.patch('semantic_release.cli.get_current_version', return_value=None)
+    with pytest.raises(ImproperConfigurationError):
+        changelog()
+
+
+@mock.patch('semantic_release.cli.debug')
+def test_main_debug(mock_debug, runner):
```
Comment (relekang) [inline tests/test_cli.py]: What is the value in this test?

File: tests/test_cli.py [hunk 4]
```diff
@@ -341,9 +473,81 @@ def test_changelog_should_call_functions(mocker, runner):
...
+    mock_previous_version.assert_called_once_with('current')
+    mock_generate_changelog.assert_called_once_with('current', None)
+    mock_markdown_changelog.assert_any_call('current', 'super changelog', header=False)
+    mock_check_token.assert_called_once_with()
+    mock_get_owner_name.assert_called_once_with()
+    mock_post_changelog.assert_called_once_with('owner', 'name', 'current',
+                                                'super md changelog')
+
+
+def test_changelog_raises_exception(mocker):
```
Comment (relekang) [inline tests/test_cli.py]: This test name should reflect what it checks, as in answer why do we expect an error here?

File: tests/test_cli.py (hunk 3 above)
Comment (fattybenji) [reply tests/test_cli.py]: hum... I think that's a work in progress that stayed there, but it actually seems to hit the lines under main when debug is enabled. I can remove it if you like.

Comment (relekang) [reply tests/test_cli.py]: Let's just leave it.

**Pull request merged**